### ISSUE TITLE
Close internal iterator for sequence returned by `to-statements` implemenation of RepositoryConnection

### ISSUE THREAD

**Issue by lkitching** [Swirrl/grafter#163]
The `to-statements` implementation of `RepositoryConnection` returns a lazy sequence backed by an internal `org.eclipse.rdf4j.repository.RepositoryResult`. This class contains a `close` which should be called when the interal sequence has been fully realised or in case of an error.

**Comment by RickMoynihan** [comment]
Agreed we should close this, but does this manifest itself as an error or a bug at all?  What symptoms does this have?

**Comment by lkitching** [comment]
I'm not aware of any issues, no.
